### DISCUSSION TITLE
tutorials/scripting/creating_script_templates

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#139] [User-contributed notes]
# tutorials/scripting/creating_script_templates

Godot provides a way to use script templates as seen in the Script Create Dialog while creating a new script: A set of built-in script templates are provided with the editor, but it is also possibl...

https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html

**Comment by NicolasLalonde** [comment]
Regarding the `node_type` in the "Template organization and naming" section: casing *does* matter for the folder name if your filesystem is case-sensitive (i.e. most linux filesystems)

**Comment by JourneyHans** [comment]
For C#, a script template is essentially a script file(.cs), where the placeholder _CLASS_ represents a class name. If I want to create multiple script templates, does that mean I have to consider resolving naming conflicts for _CLASS_? For example, using suffixes like _CLASS_A, _CLASS_B, etc. Why isn't a script template file just a simple text file?

**Reply by Calinou** [reply]
Proposal opened:

- https://github.com/godotengine/godot-proposals/issues/10960

**Comment by sweetbbak** [comment]
is there any way to add a custom script template for a custom type that will be added via a plugin?
I dont really feel comfortable editing global editor settings across other projects is appropriate and it could become messy.

**Reply by Calinou** [reply]
You can write an EditorPlugin that writes a script template file in the [project-defined script templates folder](https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html#project-defined-templates) when the plugin is enabled, and possibly remove it when the plugin is disabled.

Remember to read the project setting for the script templates folder (don't hardcode the path, as it may change across projects).

**Reply by sweetbbak** [reply]
Oh okay, thanks. That's not too bad.
